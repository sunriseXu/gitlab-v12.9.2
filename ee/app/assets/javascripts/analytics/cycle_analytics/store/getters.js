import dateFormat from 'dateformat';
import httpStatus from '~/lib/utils/http_status';
import { dateFormats } from '../../shared/constants';
import { getDurationChartData, getDurationChartMedianData, getTasksByTypeData } from '../utils';

export const hasNoAccessError = state => state.errorCode === httpStatus.FORBIDDEN;

export const currentGroupPath = ({ selectedGroup }) =>
  selectedGroup && selectedGroup.fullPath ? selectedGroup.fullPath : null;

export const selectedProjectIds = ({ selectedProjects }) =>
  selectedProjects.length ? selectedProjects.map(({ id }) => id) : [];

export const cycleAnalyticsRequestParams = ({ startDate = null, endDate = null }, getters) => ({
  project_ids: getters.selectedProjectIds,
  created_after: startDate ? dateFormat(startDate, dateFormats.isoDate) : null,
  created_before: endDate ? dateFormat(endDate, dateFormats.isoDate) : null,
});

export const durationChartPlottableData = state => {
  const { durationData, startDate, endDate } = state;
  const selectedStagesDurationData = durationData.filter(stage => stage.selected);
  const plottableData = getDurationChartData(selectedStagesDurationData, startDate, endDate);

  return plottableData.length ? plottableData : null;
};

export const durationChartMedianData = state => {
  const { durationMedianData, startDate, endDate } = state;
  const selectedStagesDurationMedianData = durationMedianData.filter(stage => stage.selected);
  const plottableData = getDurationChartMedianData(
    selectedStagesDurationMedianData,
    startDate,
    endDate,
  );

  return plottableData.length ? plottableData : [];
};

export const tasksByTypeChartData = ({ tasksByType, startDate, endDate }) => {
  if (tasksByType && tasksByType.data.length) {
    return getTasksByTypeData({
      data: tasksByType.data,
      startDate,
      endDate,
    });
  }
  return { groupBy: [], data: [], seriesNames: [] };
};

const filterStagesByHiddenStatus = (stages = [], isHidden = true) =>
  stages.filter(({ hidden = false }) => hidden === isHidden);

export const hiddenStages = ({ stages }) => filterStagesByHiddenStatus(stages);
export const activeStages = ({ stages }) => filterStagesByHiddenStatus(stages, false);
