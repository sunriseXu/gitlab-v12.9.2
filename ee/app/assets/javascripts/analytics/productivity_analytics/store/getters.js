// eslint-disable-next-line import/prefer-default-export
export const getMetricTypes = state => chartKey =>
  state.metricTypes.filter(m => m.charts.indexOf(chartKey) !== -1);
