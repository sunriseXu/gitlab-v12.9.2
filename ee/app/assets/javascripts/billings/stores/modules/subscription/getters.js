export const isFreePlan = state => state.plan.code === null;

// prevent babel-plugin-rewire from generating an invalid default during karma tests
export default () => {};
