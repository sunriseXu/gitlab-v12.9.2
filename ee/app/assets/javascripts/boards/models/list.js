/* eslint-disable no-param-reassign */
import List from '~/boards/models/list';
import ListAssignee from '~/boards/models/assignee';
import ListMilestone from '~/boards/models/milestone';

const EE_TYPES = {
  promotion: {
    isPreset: true,
    isExpandable: false,
    isBlank: true,
  },
};

class ListEE extends List {
  constructor(...args) {
    super(...args);
    this.totalWeight = 0;
  }

  getTypeInfo(type) {
    return EE_TYPES[type] || super.getTypeInfo(type);
  }

  getIssues(...args) {
    return super.getIssues(...args).then(data => {
      this.totalWeight = data.total_weight;
    });
  }

  addIssue(issue, ...args) {
    super.addIssue(issue, ...args);

    if (issue.weight) {
      this.totalWeight += issue.weight;
    }
  }

  removeIssue(issue, ...args) {
    if (issue.weight) {
      this.totalWeight -= issue.weight;
    }

    super.removeIssue(issue, ...args);
  }

  addWeight(weight) {
    this.totalWeight += weight;
  }

  onNewIssueResponse(issue, data) {
    issue.milestone = data.milestone ? new ListMilestone(data.milestone) : data.milestone;
    issue.assignees = Array.isArray(data.assignees)
      ? data.assignees.map(assignee => new ListAssignee(assignee))
      : data.assignees;
    issue.labels = data.labels;

    super.onNewIssueResponse(issue, data);
  }
}

window.List = ListEE;

export default ListEE;
