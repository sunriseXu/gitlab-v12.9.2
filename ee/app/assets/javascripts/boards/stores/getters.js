import gettersCE from '~/boards/stores/getters';

export default {
  ...gettersCE,
};
