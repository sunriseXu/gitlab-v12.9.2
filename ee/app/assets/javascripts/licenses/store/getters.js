export const hasLicenses = state => state.licenses.length > 0;

export default () => {};
