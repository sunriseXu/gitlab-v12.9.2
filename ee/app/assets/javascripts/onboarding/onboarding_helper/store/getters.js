export const stepIndex = state => {
  const { tourData, tourKey, url, projectFullPath, createdProjectPath } = state;
  let idx = -1;

  if (tourData && tourData[tourKey] && url !== '') {
    idx = tourData[tourKey].findIndex(item =>
      item.forUrl({ projectFullPath, createdProjectPath }).test(state.url),
    );
  }

  return idx !== -1 ? idx : null;
};

export const stepContent = (state, getters) => {
  const { tourData, tourKey } = state;

  if (!tourData || !tourData[tourKey] || getters.stepIndex === null) {
    return null;
  }

  return tourData[tourKey][getters.stepIndex] ? tourData[tourKey][getters.stepIndex] : null;
};

export const helpContent = (state, getters) => {
  const { projectName, helpContentIndex } = state;

  if (getters.stepContent === null) {
    return null;
  }

  return getters.stepContent.getHelpContent
    ? getters.stepContent.getHelpContent({ projectName })[helpContentIndex]
    : null;
};

export const totalTourPartSteps = state => {
  if (state.tourData && state.tourKey && state.tourData[state.tourKey]) {
    return state.tourData[state.tourKey].length;
  }

  return 0;
};

export const percentageCompleted = state => {
  const { tourData, tourKey, lastStepIndex } = state;

  if (lastStepIndex === -1 || !tourData || !tourData[tourKey]) {
    return 0;
  }

  return Math.floor((100 * lastStepIndex) / tourData[tourKey].length);
};

export const actionPopover = (state, getters) =>
  getters.stepContent !== null && getters.stepContent.actionPopover
    ? getters.stepContent.actionPopover
    : null;

// prevent babel-plugin-rewire from generating an invalid default during karma tests
export default () => {};
