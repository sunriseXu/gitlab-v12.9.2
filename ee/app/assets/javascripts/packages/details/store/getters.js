import { s__ } from '~/locale';
import { generateConanRecipe } from '../utils';
import { NpmManager } from '../constants';

export const packagePipeline = ({ packageEntity }) => {
  return packageEntity?.pipeline || null;
};

export const packageTypeDisplay = ({ packageEntity }) => {
  switch (packageEntity.package_type) {
    case 'conan':
      return s__('PackageType|Conan');
    case 'maven':
      return s__('PackageType|Maven');
    case 'npm':
      return s__('PackageType|NPM');
    case 'nuget':
      return s__('PackageType|NuGet');

    default:
      return null;
  }
};

export const conanInstallationCommand = ({ packageEntity }) => {
  const recipe = generateConanRecipe(packageEntity);

  // eslint-disable-next-line @gitlab/i18n/no-non-i18n-strings
  return `conan install ${recipe} --remote=gitlab`;
};

export const conanSetupCommand = ({ conanPath }) =>
  // eslint-disable-next-line @gitlab/i18n/no-non-i18n-strings
  `conan remote add gitlab ${conanPath}`;

export const mavenInstallationXml = ({ packageEntity = {} }) => {
  const {
    app_group: appGroup = '',
    app_name: appName = '',
    app_version: appVersion = '',
  } = packageEntity.maven_metadatum;

  return `<dependency>
  <groupId>${appGroup}</groupId>
  <artifactId>${appName}</artifactId>
  <version>${appVersion}</version>
</dependency>`;
};

export const mavenInstallationCommand = ({ packageEntity = {} }) => {
  const {
    app_group: group = '',
    app_name: name = '',
    app_version: version = '',
  } = packageEntity.maven_metadatum;

  return `mvn dependency:get -Dartifact=${group}:${name}:${version}`;
};

export const mavenSetupXml = ({ mavenPath }) => `<repositories>
  <repository>
    <id>gitlab-maven</id>
    <url>${mavenPath}</url>
  </repository>
</repositories>

<distributionManagement>
  <repository>
    <id>gitlab-maven</id>
    <url>${mavenPath}</url>
  </repository>

  <snapshotRepository>
    <id>gitlab-maven</id>
    <url>${mavenPath}</url>
  </snapshotRepository>
</distributionManagement>`;

export const npmInstallationCommand = ({ packageEntity }) => (type = NpmManager.NPM) => {
  // eslint-disable-next-line @gitlab/i18n/no-non-i18n-strings
  const instruction = type === NpmManager.NPM ? 'npm i' : 'yarn add';

  return `${instruction} ${packageEntity.name}`;
};

export const npmSetupCommand = ({ packageEntity, npmPath }) => (type = NpmManager.NPM) => {
  const scope = packageEntity.name.substring(0, packageEntity.name.indexOf('/'));

  if (type === NpmManager.NPM) {
    return `echo ${scope}:registry=${npmPath} >> .npmrc`;
  }

  return `echo \\"${scope}:registry\\" \\"${npmPath}\\" >> .yarnrc`;
};

export const nugetInstallationCommand = ({ packageEntity }) =>
  `nuget install ${packageEntity.name} -Source "GitLab"`;

export const nugetSetupCommand = ({ nugetPath }) =>
  `nuget source Add -Name "GitLab" -Source "${nugetPath}" -UserName <your_username> -Password <your_token>`;
