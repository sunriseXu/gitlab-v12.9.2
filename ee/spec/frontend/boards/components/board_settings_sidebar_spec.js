import '~/boards/models/list';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { GlDrawer, GlLabel, GlFormInput, GlAvatarLink, GlAvatarLabeled } from '@gitlab/ui';
import BoardSettingsSidebar from 'ee/boards/components/board_settings_sidebar.vue';
import boardsStore from 'ee_else_ce/boards/stores/boards_store_ee';
import getters from 'ee_else_ce/boards/stores/getters';
import bs from '~/boards/stores/boards_store';
import flash from '~/flash';
import waitForPromises from 'helpers/wait_for_promises';

jest.mock('~/flash');
// NOTE: needed for calling boardsStore.addList

const localVue = createLocalVue();

localVue.use(Vuex);

describe('BoardSettingsSideBar', () => {
  let wrapper;
  let mock;
  let storeActions;
  const labelTitle = 'test';
  const labelColor = '#FFFF';
  const listId = 1;
  const currentWipLimit = 1; // Needs to be other than null to trigger requests.

  const createComponent = (state = { activeListId: 0 }, actions = {}, localState = {}) => {
    storeActions = actions;

    const store = new Vuex.Store({
      state,
      actions: storeActions,
      getters,
    });

    wrapper = shallowMount(BoardSettingsSidebar, {
      store,
      localVue,
      data() {
        return localState;
      },
    });
  };

  const triggerBlur = type => {
    if (type === 'blur') {
      wrapper.find(GlFormInput).vm.$emit('blur');
    }

    if (type === 'enter') {
      wrapper.find(GlFormInput).trigger('keydown.enter');
    }
  };

  beforeEach(() => {
    // mock CE store
    const storeMock = {
      state: { lists: [] },
      create() {},
      setCurrentBoard: jest.fn(),
      findList: bs.findList,
      addList: bs.addList,
      removeList: bs.removeList,
      scopedLabels: {
        enabled: false,
        scopedLabelsDocumentationLink: '',
      },
    };

    boardsStore.initEESpecific(storeMock);
  });

  afterEach(() => {
    jest.restoreAllMocks();
    wrapper.destroy();
  });

  describe('GlDrawer', () => {
    it('finds a GlDrawer component', () => {
      createComponent();

      expect(wrapper.find(GlDrawer).exists()).toBe(true);
    });

    describe('on close', () => {
      it('calls closeSidebar', () => {
        const spy = jest.fn();
        createComponent({ activeListId: 0 }, { setActiveListId: spy });

        wrapper.find(GlDrawer).vm.$emit('close');

        return wrapper.vm.$nextTick().then(() => {
          expect(storeActions.setActiveListId).toHaveBeenCalledWith(
            expect.anything(),
            0,
            undefined,
          );
        });
      });
    });

    describe('when activeListId is zero', () => {
      it('renders GlDrawer with open false', () => {
        createComponent();

        expect(wrapper.find(GlDrawer).props('open')).toBe(false);
      });
    });

    describe('when activeListId is greater than zero', () => {
      beforeEach(() => {
        mock = new MockAdapter(axios);

        boardsStore.store.addList({
          id: listId,
          label: { title: labelTitle, color: labelColor },
          list_type: 'label',
        });
      });

      afterEach(() => {
        boardsStore.store.removeList(listId);
      });

      it('renders GlDrawer with open false', () => {
        createComponent({ activeListId: 1 });

        expect(wrapper.find(GlDrawer).props('open')).toBe(true);
      });
    });

    describe('when activeListId is in boardsStore', () => {
      beforeEach(() => {
        mock = new MockAdapter(axios);

        boardsStore.store.addList({
          id: listId,
          label: { title: labelTitle, color: labelColor },
          list_type: 'label',
        });

        createComponent({ activeListId: listId });
      });

      afterEach(() => {
        mock.restore();
      });

      it('renders label title', () => {
        expect(wrapper.find(GlLabel).props('title')).toEqual(labelTitle);
      });

      it('renders label background color', () => {
        expect(wrapper.find(GlLabel).props('backgroundColor')).toEqual(labelColor);
      });
    });

    describe('when activeListId is not in boardsStore', () => {
      beforeEach(() => {
        mock = new MockAdapter(axios);

        boardsStore.store.addList({ id: listId, label: { title: labelTitle, color: labelColor } });

        createComponent({ activeListId: 0 });
      });

      afterEach(() => {
        mock.restore();
      });

      it('does not render GlLabel', () => {
        expect(wrapper.find(GlLabel).exists()).toBe(false);
      });
    });
  });

  describe('when activeList is present', () => {
    beforeEach(() => {
      mock = new MockAdapter(axios);
    });

    afterEach(() => {
      boardsStore.store.removeList(listId);
    });

    describe('when activeListWipLimit is 0', () => {
      beforeEach(() => {
        boardsStore.store.addList({
          id: listId,
          label: { title: labelTitle, color: labelColor },
          max_issue_count: 0,
          list_type: 'label',
        });
      });

      it('renders "None" in the block', () => {
        createComponent({ activeListId: listId });

        expect(wrapper.find('.js-wip-limit').text()).toMatchSnapshot();
      });
    });

    describe('when activeListWipLimit is greater than 0', () => {
      describe('when list type is "milestone"', () => {
        beforeEach(() => {
          boardsStore.store.addList({
            id: 1,
            milestone: {
              webUrl: 'https://gitlab.com/h5bp/html5-boilerplate/-/milestones/1',
              title: 'Backlog',
            },
            max_issue_count: 1,
            list_type: 'milestone',
          });
        });

        afterEach(() => {
          boardsStore.store.removeList(1, 'milestone');
          wrapper.destroy();
        });

        it('renders the correct milestone text', () => {
          createComponent({ activeListId: 1 });

          expect(wrapper.find('.js-milestone').text()).toMatchSnapshot();
        });

        it('renders the correct list type text', () => {
          createComponent({ activeListId: 1 });

          expect(wrapper.find('.js-list-label').text()).toMatchSnapshot();
        });
      });

      describe('when list type is "assignee"', () => {
        beforeEach(() => {
          boardsStore.store.addList({
            id: 1,
            user: { username: 'root', avatar: '', name: 'Test', webUrl: 'https://gitlab.com/root' },
            max_issue_count: 1,
            list_type: 'assignee',
          });
        });

        afterEach(() => {
          boardsStore.store.removeList(1, 'assignee');
          wrapper.destroy();
        });

        it('renders gl-avatar-link with correct href', () => {
          createComponent({ activeListId: 1 });

          expect(wrapper.find(GlAvatarLink).exists()).toBe(true);
          expect(wrapper.find(GlAvatarLink).attributes('href')).toEqual('https://gitlab.com/root');
        });

        it('renders gl-avatar-labeled with "root" as username and name as "Test"', () => {
          createComponent({ activeListId: 1 });

          expect(wrapper.find(GlAvatarLabeled).exists()).toBe(true);
          expect(wrapper.find(GlAvatarLabeled).attributes('label')).toEqual('Test');
          expect(wrapper.find(GlAvatarLabeled).attributes('sublabel')).toEqual('@root');
        });

        it('renders the correct list type text', () => {
          createComponent({ activeListId: 1 });

          expect(wrapper.find('.js-list-label').text()).toMatchSnapshot();
        });
      });

      it.each`
        num
        ${1}
        ${11}
      `('it renders $num', ({ num }) => {
        boardsStore.store.addList({
          id: num,
          label: { title: labelTitle, color: labelColor },
          max_issue_count: num,
          list_type: 'label',
        });

        createComponent({ activeListId: num });

        expect(wrapper.find('.js-wip-limit').text()).toMatchSnapshot();
      });
    });
  });

  describe('when clicking edit', () => {
    beforeEach(() => {
      mock = new MockAdapter(axios);
      boardsStore.store.addList({
        id: listId,
        label: { title: labelTitle, color: labelColor },
        max_issue_count: 4,
        list_type: 'label',
      });

      createComponent({ activeListId: listId }, { updateListWipLimit: () => {} });
    });

    it('renders an input', () => {
      wrapper.find('.js-edit-button').vm.$emit('click');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.find(GlFormInput).exists()).toBe(true);
      });
    });

    it('does not render current wipLimit text', () => {
      wrapper.find('.js-edit-button').vm.$emit('click');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.find('.js-wip-limit').exists()).toBe(false);
      });
    });

    it('sets wipLimit to be the value of list.maxIssueCount', () => {
      expect(wrapper.vm.currentWipLimit).toEqual(null);

      wrapper.find('.js-edit-button').vm.$emit('click');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.vm.currentWipLimit).toBe(4);
      });
    });
  });

  describe('remove limit', () => {
    describe('when wipLimit is set', () => {
      beforeEach(() => {
        mock = new MockAdapter(axios);
        boardsStore.store.addList({
          id: listId,
          label: { title: labelTitle, color: labelColor },
          max_issue_count: 4,
          list_type: 'label',
        });
        const spy = jest.fn().mockResolvedValue({
          config: { data: JSON.stringify({ list: { max_issue_count: 0 } }) },
        });

        createComponent({ activeListId: listId }, { updateListWipLimit: spy });
      });

      it('resets wipLimit to 0', () => {
        expect(wrapper.vm.activeList.maxIssueCount).toEqual(4);

        wrapper.find('.js-remove-limit').vm.$emit('click');

        return wrapper.vm.$nextTick().then(() => {
          expect(wrapper.vm.activeList.maxIssueCount).toEqual(0);
        });
      });
    });

    describe('when wipLimit is not set', () => {
      beforeEach(() => {
        mock = new MockAdapter(axios);
        boardsStore.store.addList({
          id: listId,
          label: { title: labelTitle, color: labelColor },
          max_issue_count: 0,
          list_type: 'label',
        });

        createComponent({ activeListId: listId }, { updateListWipLimit: () => {} });
      });

      it('does not render the remove limit button', () => {
        expect(wrapper.find('.js-remove-limit').exists()).toBe(false);
      });
    });
  });

  describe('when edit is true', () => {
    beforeEach(() => {
      mock = new MockAdapter(axios);

      boardsStore.store.addList({
        id: listId,
        label: { title: labelTitle, color: labelColor },
        max_issue_count: 2,
        list_type: 'label',
      });
    });

    afterEach(() => {
      flash.mockReset();
      boardsStore.store.removeList(listId, 'label');
    });

    describe.each`
      blurMethod
      ${'enter'}
      ${'blur'}
    `('$blurMethod', ({ blurMethod }) => {
      describe(`when blur is triggered by ${blurMethod}`, () => {
        it('calls updateListWipLimit', () => {
          const spy = jest.fn().mockResolvedValue({
            config: { data: JSON.stringify({ list: { max_issue_count: '4' } }) },
          });
          createComponent(
            { activeListId: 1 },
            { updateListWipLimit: spy },
            { edit: true, currentWipLimit },
          );

          triggerBlur(blurMethod);

          return wrapper.vm.$nextTick().then(() => {
            expect(spy).toHaveBeenCalledTimes(1);
          });
        });

        describe('when component wipLimit and List.maxIssueCount are equal', () => {
          it('doesnt call updateListWipLimit', () => {
            const spy = jest.fn().mockResolvedValue({});
            createComponent(
              { activeListId: 1 },
              { updateListWipLimit: spy },
              { edit: true, currentWipLimit: 2 },
            );

            triggerBlur(blurMethod);

            return wrapper.vm.$nextTick().then(() => {
              expect(spy).toHaveBeenCalledTimes(0);
            });
          });
        });

        describe('when currentWipLimit is null', () => {
          it('doesnt call updateListWipLimit', () => {
            const spy = jest.fn().mockResolvedValue({});
            createComponent(
              { activeListId: 1 },
              { updateListWipLimit: spy },
              { edit: true, currentWipLimit: null },
            );

            triggerBlur(blurMethod);

            return wrapper.vm.$nextTick().then(() => {
              expect(spy).toHaveBeenCalledTimes(0);
            });
          });
        });

        describe('when response is successful', () => {
          const maxIssueCount = 11;

          beforeEach(() => {
            const spy = jest.fn().mockResolvedValue({});
            createComponent(
              { activeListId: 1 },
              { updateListWipLimit: spy },
              { edit: true, currentWipLimit: maxIssueCount },
            );

            triggerBlur(blurMethod);

            return waitForPromises();
          });

          it('sets activeWipLimit to new maxIssueCount value', () => {
            /*
             * DANGER: bad coupling to the computed prop of the component because the
             * computed prop relys on the list from boardStore, for now this is the way around
             * stale values from boardsStore being updated, when we move List and BoardsStore to Vuex
             * or Graphql we will be able to query the DOM for the new value.
             */
            expect(wrapper.vm.activeList.maxIssueCount).toEqual(maxIssueCount);
          });

          it('toggles GlFormInput on blur', () => {
            expect(wrapper.find(GlFormInput).exists()).toBe(false);
            expect(wrapper.find('.js-wip-limit').exists()).toBe(true);
            expect(wrapper.vm.updating).toBe(false);
          });
        });

        describe('when response fails', () => {
          beforeEach(() => {
            const spy = jest.fn().mockRejectedValue();
            createComponent(
              { activeListId: 1 },
              { updateListWipLimit: spy, setActiveListId: () => {} },
              { edit: true, currentWipLimit },
            );

            triggerBlur(blurMethod);

            return waitForPromises();
          });

          it('calls flash with expected error', () => {
            expect(flash).toHaveBeenCalledTimes(1);
          });
        });
      });
    });

    describe('passing of props to gl-form-input', () => {
      beforeEach(() => {
        createComponent({ activeListId: listId }, { updateListWipLimit: () => {} }, { edit: true });
      });

      it('passes `trim`', () => {
        expect(wrapper.find(GlFormInput).attributes().trim).toBeDefined();
      });

      it('passes `number`', () => {
        expect(wrapper.find(GlFormInput).attributes().number).toBeDefined();
      });
    });
  });
});
