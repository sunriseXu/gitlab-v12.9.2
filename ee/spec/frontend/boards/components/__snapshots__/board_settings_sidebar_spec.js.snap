// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardSettingsSideBar when activeList is present when activeListWipLimit is 0 renders "None" in the block 1`] = `"None"`;

exports[`BoardSettingsSideBar when activeList is present when activeListWipLimit is greater than 0 it renders 1 1`] = `"1 issue"`;

exports[`BoardSettingsSideBar when activeList is present when activeListWipLimit is greater than 0 it renders 11 1`] = `"11 issues"`;

exports[`BoardSettingsSideBar when activeList is present when activeListWipLimit is greater than 0 when list type is "assignee" renders the correct list type text 1`] = `"Assignee"`;

exports[`BoardSettingsSideBar when activeList is present when activeListWipLimit is greater than 0 when list type is "milestone" renders the correct list type text 1`] = `"Milestone"`;

exports[`BoardSettingsSideBar when activeList is present when activeListWipLimit is greater than 0 when list type is "milestone" renders the correct milestone text 1`] = `"Backlog"`;
